### PR TITLE
refactor(src/run.js): use spawn instead of exec

### PR DESCRIPTION
**Pull request by DeclanBoller**
In a large monorepo the output from running some git commands can be huge. The default max size for stdout that child_process.exec can handle is 200KB, we 'hacked' this for a quickfix in our repo but the ling term fix is `spawn`. This PR leverages off the work done in #277. 

This should address the issue raised here: https://github.com/ember-cli/ember-cli-update/issues/583

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
 
-module.exports = async function run(command, options) {
+const run = async function run(command, options = {}) {
+  // Matches the shell-enabled behaviour of child_process.exec, which this
+  // function previously used to run the specified command. This avoids
+  // problems when handling quotes and whitespace.
+  options.shell = true;
   debug(command);
-  let { stdout } = await execPromise(command, options);
+  let [cmd, ...args] = command.split(' ');
```
Comment (kellyselden) [inline src/run.js]: Will `--foo="bar baz"` still get processed correctly with this split?

Comment (timiyay) [reply src/run.js]: Hmmm I believe it won't. It'll split inside the quotes.

@DeclanBoller we'll need to take another pass at this, improving the split logic to handle this case.

**Comment by DeclanBoller** [comment]
Is this likely to still be an issue that needs to be addressed? Noticed that you closed the two issues (#583 & #283) that triggered the creation of this PR. @kellyselden

**Comment by kellyselden** [comment]
This should be resolved as of https://github.com/kellyselden/git-diff-apply/pull/370. Thanks for your help!

**Pull request closed without merging**